sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/processor.hpp

```diff
@@ -17,206 +17,244 @@ namespace Azure { namespace Messaging { namespace EventHubs { namespace Test {
 }}}} // namespace Azure::Messaging::EventHubs::Test
 #endif
 namespace Azure { namespace Messaging { namespace EventHubs {
+/**@brief ProcessorOptions are the options for the CreateProcessor
+ * function.
+ */
+struct ProcessorOptions final
+{
+  /**@brief LoadBalancingStrategy dictates how concurrent Processor instances distribute
+   * ownership of partitions between them.
+   * The default strategy is ProcessorStrategyBalanced.
+   */
+  Models::ProcessorStrategy LoadBalancingStrategy;
+
+  /**@brief UpdateInterval controls how often attempt to claim partitions.
+   * The default value is 10 seconds.
+   */
+  Azure::DateTime::duration UpdateInterval;
 
-  /**@brief Processor uses a [ConsumerClient] and [CheckpointStore] to provide automatic
-   * load balancing between multiple Processor instances, even in separate
-   *processes or on separate machines.
+  /**@brief PartitionExpirationDuration is the amount of time before a partition is
+   * considered unowned. The default value is 60 seconds.
    */
-  class Processor {
+  Azure::DateTime::duration PartitionExpirationDuration;
+
+  /**@brief StartPositions are the default start positions (configurable per
+   * partition, or with an overall default value) if a checkpoint is not found
+   * in the CheckpointStore. The default position is Latest.
+   */
+  Models::StartPositions StartPositions;
+
+  /**@brief Prefetch represents the size of the internal prefetch buffer for
+   * each ProcessorPartitionClient created by this Processor. When
+   * set, this client will attempt to always maintain an internal
+   * cache of events of this size, asynchronously, increasing the odds
+   * that ReceiveEvents() will use a locally stored cache of events,
+   * rather than having to wait for events to arrive from the network.
+   *
+   * Defaults to 300 events.
+   * Disabled if Prefetch < 0.
+   */
+  int32_t Prefetch = 300;
+};
+
+/**@brief Processor uses a [ConsumerClient] and [CheckpointStore] to provide automatic
+ * load balancing between multiple Processor instances, even in separate
+ *processes or on separate machines.
+ */
+class Processor final {
```

Comment:
If I read correctly, .NET SDK has this class as internal.
They have `EventProcessorClient` which is public, but this one is internal?
This is not for this PR, but for the ApiView phase - please when reviewing the API, think in the aspect of whether we can reduce the amount of abstractions that we expose publicly.

Reply:
The C++ SDK is based off the constructs in the Go SDK (since that is the most recent EventHubs SDK). And the Processor is basically not functional at this point.